### PR TITLE
Use this.props.state as previousState for first action response

### LINKED ISSUE #27: Getting "Error selecting state." when using custom selector
When I try to use a custom state selector, I get "Error selecting state." printed under `@@INIT` action. It appears state is null during this action, because I can't even select a top level property, e.g. `state => state.propA`. Is this by design?

### PR DESCRIPTION
**Pull request by danny-andrews**
Fixes #27.

### CODE REVIEW

**Comment by danny-andrews** [comment]
Don't accept this yet. I found the root of the problem. this.props.previousState starts out as null.

**Comment by danny-andrews** [comment]
There we go. :)

File: src/LogMonitorEntry.js [hunk 1]
```diff
@@ -31,6 +31,7 @@ export default class LogMonitorEntry extends Component {
   constructor(props) {
     super(props);
     this.handleActionClick = this.handleActionClick.bind(this);
+    this.previousState = this.props.previousState || this.props.state;
```
Comment (gaearon) [inline src/LogMonitorEntry.js]: This is incorrect. Component can receive new props in `componentWillReceiveProps` so it is wrong to calculate anything based on props in the constructor and cache it.

Comment (gaearon) [inline src/LogMonitorEntry.js]: In general, this seems to give you the wrong semantics. You are making it look like the previous state *is* the current state for the first action. However what really happens is that it doesn’t exist.

File: src/LogMonitorEntry.js [hunk 2]
```diff
@@ -42,7 +43,7 @@ export default class LogMonitorEntry extends Component {
             theme={this.props.theme}
             keyName={'state'}
             data={this.props.select(state)}
-            previousData={this.props.select(this.props.previousState)}
+            previousData={this.props.select(this.previousState)}
```
Comment (gaearon) [inline src/LogMonitorEntry.js]: Why not `previousData={typeof this.props.state !== 'undefined' ? this.props.select(this.props.state) : undefined}`?

**Comment by gaearon** [comment]
Thank you again for the PR and taking a look at this.
I applied what I believe to be a more correct fix in 3af3102c48102296b82aacfcc9dc445b4df80b40.

**Pull request closed without merging**